core/perf_test/test_taskdag.cpp

```diff
@@ -226,6 +226,8 @@ int main(int argc, char* argv[]) {
   return 0;
 }
 
+#ifdef KOKKOS_ENABLE_DEPRECATION_WARNINGS
```

Comment:
It looks weird that we enable deprecation warnings just to disable them in next line. Can you please elaborate? Also I am not sure if I see the matching `KOKKOS_IMPL_DISABLE_DEPRECATED_WARNINGS_PUSH()`

Reply:
`KOKKOS_ENABLE_DEPRECARTION_WARNINGS` makes `KOKKOS_DEPRECATED` expand to `[[deprecated]]`.
`KOKKOS_IMPL_DISABLE_DEPRECATED_WARNINGS_PUSH/POP` however suppresses compiler warnings like `-Wdeprecated-declarations` and thus covers more than just the deprecation warnings coming from `Kokkos`.

Reply:
> Also I am not sure if I see the matching `KOKKOS_IMPL_DISABLE_DEPRECATED_WARNINGS_PUSH()`

Fixed in fb7810747.